src/plugins/intel_npu/tests/functional/behavior/remote_run.hpp

```diff
@@ -0,0 +1,401 @@
+namespace behavior {
+class RemoteRunTests : public ov::test::behavior::OVPluginTestBase,
+                       public testing::WithParamInterface<CompilationParams> {
+protected:
+    std::shared_ptr<ov::Core> core = utils::PluginCache::get().core();
+    ov::AnyMap configuration;
+    std::shared_ptr<ov::Model> ov_model;
+    ov::CompiledModel compiled_model;
+    int _fd_dma_heap = -1;
+
+public:
+    static std::string getTestCaseName(testing::TestParamInfo<CompilationParams> obj) {
+        std::string targetDevice;
+        ov::AnyMap configuration;
+        std::tie(targetDevice, configuration) = obj.param;
+        std::replace(targetDevice.begin(), targetDevice.end(), ':', '_');
+        targetDevice = ov::test::utils::getTestsPlatformFromEnvironmentOr(ov::test::utils::DEVICE_NPU);
+
+        std::ostringstream result;
+        result << "targetDevice=" << targetDevice << "_";
+        result << "targetPlatform=" << ov::test::utils::getTestsPlatformFromEnvironmentOr(targetDevice) << "_";
+        if (!configuration.empty()) {
+            for (auto& configItem : configuration) {
+                result << "configItem=" << configItem.first << "_";
+                configItem.second.print(result);
+            }
+        }
+
+        return result.str();
+    }
+
+    void SetUp() override {
+        std::tie(target_device, configuration) = this->GetParam();
+
+        SKIP_IF_CURRENT_TEST_IS_DISABLED()
+        OVPluginTestBase::SetUp();
+        ov_model = getDefaultNGraphFunctionForTheDeviceNPU();  // FIXME: E#80555
+    }
+
+    void TearDown() override {
+        if (!configuration.empty()) {
+            utils::PluginCache::get().reset();
+        }
+
+#ifdef __linux__
+#    if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 10, 0)
+        if (_fd_dma_heap != -1) {
+            close(_fd_dma_heap);
+        }
+#    endif
+#endif
+
+        APIBaseTest::TearDown();
+    }
+
+#ifdef __linux__
+#    if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 10, 0)
+    int32_t getFdDmaHeap(const size_t byte_size) {
+        if (!std::filesystem::exists("/dev/dma_heap/system")) {
+            OPENVINO_THROW("Cannot open /dev/dma_heap/system file.");
+        }
+
+        _fd_dma_heap = open("/dev/dma_heap/system", O_RDWR);
+        if (_fd_dma_heap == -1) {
+            OPENVINO_THROW("Cannot open /dev/dma_heap/system.");
+        }
+
+        static const std::size_t alignment = 4096;
+        size_t size = byte_size + alignment - (byte_size % alignment);
+        struct dma_heap_allocation_data heapAlloc = {
+            .len = size,  // this length should be alligned to the page size
+            .fd = 0,
+            .fd_flags = O_RDWR | O_CLOEXEC,
+            .heap_flags = 0,
+        };
+
+        auto ret = ioctl(_fd_dma_heap, DMA_HEAP_IOCTL_ALLOC, &heapAlloc);
+        if (ret != 0) {
+            OPENVINO_THROW("Cannot initialize DMA heap");
+        }
+
+        return static_cast<int32_t>(heapAlloc.fd);
+    }
+#    endif
+#endif
+};
+
```

Comment:
The tests here cover host allocated memory (with `new`) that is not visible by NPU. I suggest we also add (Windows) tests that use a) custom heap for CPU-NPU sharing, and b) cross adapter heap for GPU-NPU sharing.

Reply:
To be honest not sure that I understand this comment. I am not sure how to add Windows tests here since we need a third-party sources(DirectX)

Reply:
Added the tests for DirectX but they can only be run locally for now. Too many dependencies to run them on CI